xcode/Subconscious/Shared/Components/AddressBook/AddressBookView.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  AddressBookView.swift
+//  Subconscious (iOS)
+//
+//  Created by Ben Follington on 2/27/23.
+//
+
+import SwiftUI
+import ObservableStore
+
+struct AddressBookEntry {
+    var pfp: Image
+    var petname: String
+    var did: String
+}
+
+struct AddressBookView: View {
+    @ObservedObject var app: Store<AppModel>
+    var unknown = "Unknown"
+    
+    @State var searchText = ""
+    
+    // Temp state, will be moved to slice of AppStore later
+    @State var didCopy = false
+    
+    // Temp state, will be moved to slice of AppStore later
+    @State var following: [AddressBookEntry] = [
+        AddressBookEntry(pfp: Image("pfp-dog"), petname: "ben", did: "did:key:z6MkmCJAZansQ3p1Qwx6wrF4c64yt2rcM8wMrH5Rh7DGb2K7"),
+        AddressBookEntry(pfp: Image("sub_logo_light"), petname: "bob", did: "did:key:z6MkmBJAZansQ3p1Qwx6wrF4c64yt2rcM8wMrH5Rh7DGb2K7"),
+        AddressBookEntry(pfp: Image("sub_logo_dark"), petname: "alice", did: "did:key:z6MjmBJAZansQ3p1Qwx6wrF4c64yt2rcM8wMrH5Rh7DGb2K7")
+    ]
+    
+    var myDid = "did:key:z6MkmCJAZansQ3p1Qwx6wrF4c64yt2rcM8wMrH5Rh7DGb2K7"
+    
+    func delete(at offsets: IndexSet) {
+        following.remove(atOffsets: offsets)
+    }
+    
+    func filteredFollowing() -> [AddressBookEntry] {
+        return following.filter { user in
+            if (searchText.count > 0) {
+                return user.petname.lowercased().range(of: searchText.lowercased()) != nil
+            } else {
+                return true
+            }
+        }
+    }
+
+    var body: some View {
+        NavigationStack {
+            Form {
+                if (following.count == 0) {
+                    Text("No friends?")
+                } else {
+                    Section(header: Text("Friends")) {
+                        List {
+                            ForEach(filteredFollowing(), id: \.did) { user in
+                                AddressBookEntryView(pfp: user.pfp, petname: user.petname,
+                                                     did: user.did)
+                            }
+                            .onDelete(perform: delete)
+                        }
+                        .searchable(text: $searchText)
```

Comment:
For some reason adding `searchable` prevents `onDelete` from working here, the swipe-to-delete behaviour is cancelled by search. Quick google revealed nothing, but I assume there's a way to get this to work.